presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -559,72 +559,102 @@ std::unique_ptr<TaskInfo> TaskManager::createOrUpdateTaskImpl(
   PromiseHolderWeakPtr<std::unique_ptr<protocol::TaskStatus>> statusRequest;
   PromiseHolderWeakPtr<std::unique_ptr<protocol::TaskInfo>> infoRequest;
 
-  // Create or update task can be called concurrently for the same task.
-  // We need to lock here for allow only one to be executed at a time.
-  // This is especially important for adding splits to the task.
-  std::lock_guard<std::mutex> l(prestoTask->mutex);
+  bool startNextQueuedTask = false;
+  std::unique_ptr<TaskInfo> ret;
```

Comment:
```
nit: s/ret/taskInfo/ move to the place where it is first used?
```

Reply:
We cannot do this.
The ret is returned from this method and is initialized inside the scope with a lock.
So it has to be defined before the scope.